### PR TITLE
Patches recovered from old installation 2

### PR DESCRIPTION
**Pull request by andersmarkussen**

### CODE REVIEW

File: root_profiler/sheets/standard/Serenity.php [hunk 1]
```diff
@@ -62,7 +62,7 @@
 <!-- Ugly Internet Explorer Hack -->
 <?php
 $agent = $_SERVER['HTTP_USER_AGENT'];
-$firefox = eregi("Firefox", $agent);
+$firefox = preg_match('/Firefox/i', $agent);
 if( $firefox ) { echo '<!--'; } ?>
     <table id="maintable">
       <tr>
```
Comment (julien-picalausa) [inline root_profiler/sheets/standard/Serenity.php]: Ideally, we should just get rid of these "Ugly hacks" Internet Explorer is not relevant anymore, and it looks like that hack would affect any chromium-based browsers too.

File: root_profiler/engine/campaign.class.php [hunk 2]
```diff
@@ -47,6 +47,10 @@ function Campaign($id = 0)
         $this->_valid = true;
       }
     }
+    function Campaign($id = 0)
+    {
+        self::__construct();
+    }
```
Comment (julien-picalausa) [inline root_profiler/engine/campaign.class.php]: Is there any reason other than keeping PHP4 compatibility for keeping old-style constuctors around?

Comment (andersmarkussen) [reply root_profiler/engine/campaign.class.php]: Nope. This was intentionally a "least disrupion possible" push.

Comment (julien-picalausa) [reply root_profiler/engine/campaign.class.php]: Then I'd say just get rid of them. I don't think anyone sane should be running PHP4 anymore.

File: root_profiler/sheets/standard/Serenity.php (hunk 1 above)
Comment (andersmarkussen) [reply root_profiler/sheets/standard/Serenity.php]: I frankly agree. 10-year-old browser-specific workarounds are going to be utter shite. I was too focused on resolving the conflict to see the deeper issue.

**Pull request merged**